src/main/kotlin/org/rust/lang/core/psi/RustCompositeElement.kt

```diff
@@ -10,3 +13,21 @@ interface RustCompositeElement : PsiElement {
 
 val RustCompositeElement.containingMod: RustMod?
     get() = PsiTreeUtil.getStubOrPsiParentOfType(this, RustMod::class.java)
+
+/**
+ * Returns the level of the given lint for this element.
+ */
+fun PsiElement.lintLevel(lint: RustLint): RustLintLevel {
+    var level: RustLintLevel? = null
+    if (this is RustDocAndAttributeOwner) {
+        level = queryAttributes.metaItems
+            .filter { it.metaItemList.any { it.text == lint.id } }
+            .map { RustLintLevel.valueForId(it.identifier.text) }
+            .filterNotNull()
+            .firstOrNull()
+    }
+    if (level == null) {
+        level = parentOfType<RustDocAndAttributeOwner>()?.lintLevel(lint)
```

Comment:
No, there's no API for iterating both inside and outside a file. There is `RustMod.superMods` however which should help.